relay-ourlogs/src/vercel_to_sentry.rs

```diff
@@ -0,0 +1,722 @@
+    /// Action taken by firewall rules.
+    #[serde(rename = "wafAction")]
+    pub waf_action: Option<String>,
+    /// ID of the firewall rule that matched.
+    #[serde(rename = "wafRuleId")]
+    pub waf_rule_id: Option<String>,
+}
+
+/// Vercel log structure matching their schema.
+/// Based on: https://vercel.com/docs/drains/reference/logs
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct VercelLog {
+    /// Unique identifier for the log entry.
+    pub id: String,
+    /// Identifier for the Vercel deployment.
+    #[serde(rename = "deploymentId")]
+    pub deployment_id: String,
+    /// Origin of the log.
+    pub source: String,
+    /// Deployment unique URL hostname.
+    pub host: String,
+    /// Unix timestamp in milliseconds.
+    pub timestamp: i64,
+    /// Identifier for the Vercel project.
+    #[serde(rename = "projectId")]
+    pub project_id: String,
+    // Log severity level.
+    pub level: VercelLogLevel,
+    /// Log message content (may be truncated if over 256 KB).
+    pub message: Option<String>,
+    /// Identifier for the Vercel build. Only present on build logs.
+    #[serde(rename = "buildId")]
+    pub build_id: Option<String>,
+    /// Entrypoint for the request.
+    pub entrypoint: Option<String>,
+    /// Origin of the external content. Only on external logs.
+    pub destination: Option<String>,
+    /// Function or dynamic path of the request.
+    pub path: Option<String>,
+    /// Log output type.
+    #[serde(rename = "type")]
+    pub log_type: Option<String>,
+    /// Log output type.
+    #[serde(rename = "statusCode")]
+    pub status_code: Option<i64>,
+    /// Identifier of the request.
+    #[serde(rename = "requestId")]
+    pub request_id: Option<String>,
+    /// Deployment environment. One of production or preview.
+    pub environment: Option<String>,
+    /// Git branch name.
+    pub branch: Option<String>,
+    /// JA3 fingerprint digest.
+    #[serde(rename = "ja3Digest")]
+    pub ja3_digest: Option<String>,
+    /// JA4 fingerprint digest.
+    #[serde(rename = "ja4Digest")]
+    pub ja4_digest: Option<String>,
+    /// Type of edge runtime. One of edge-function or middleware
+    #[serde(rename = "edgeType")]
+    pub edge_type: Option<String>,
+    /// Name of the Vercel project.
+    #[serde(rename = "projectName")]
+    pub project_name: Option<String>,
+    /// Region where the request is executed.
+    #[serde(rename = "executionRegion")]
+    pub execution_region: Option<String>,
+    /// Trace identifier for distributed tracing.
+    #[serde(rename = "traceId")]
+    pub trace_id: Option<String>,
+    /// Span identifier for distributed tracing.
+    #[serde(rename = "spanId")]
+    pub span_id: Option<String>,
+    /// Trace identifier for distributed tracing.
+    #[serde(rename = "trace.id")]
+    pub trace_dot_id: Option<String>,
+    /// Span identifier for distributed tracing.
+    #[serde(rename = "span.id")]
+    pub span_dot_id: Option<String>,
```

Comment:
You can use [serde alias](https://serde.rs/field-attrs.html#alias) to prevent having multiple fields here.

Reply:
In that case having two fields makes sense. But what's the point in keeping both versions on serialization? Shouldn't we normalize the field into the more common name?